Sources/ProjectDescription/SourceFilesList.swift

```diff
@@ -57,6 +67,30 @@ public struct SourceFileGlob: Codable, Equatable, Sendable {
             compilationCondition: compilationCondition
         )
     }
+
+    /// Returns a source generated source file configuration, for a single generated file.
+    ///
+    /// - Parameters:
+    ///   - path: Path to the generated file. Assumed to be a specific path (as oppose to a glob pattern).
+    ///   - compilerFlags: The compiler flags to be set to the source files in the sources build phase.
+    ///   - codeGen: The source file attribute to be set in the build phase.
+    ///   - compilationCondition: Condition for file compilation.
+
+    public static func targetScriptGenerated(
```

Comment:
What do you think of naming this `generated` only? I understand the main purpose here is for scripts, but from Tuist's perspective, it doesn't really matter how exactly these files are generated – what's important is that they are always included in the generated Xcode project.

If so, then `GenerationSource` should also be refactored to something like:
```swift
public var generated: Bool
```

or:
```swift
public var type: SourceFileType

enum SourceFileType {
   /// Always present on the disk before generating a project
  case alwaysPresent
  /// Might not be present when generating a project and can be generated later on
  case generated
```

Reply:
My thinking was that if we know it came from scripts, we could in the future do things like verify that it's listed as an output path of some script. And if there will be another source of generation that's not from script, there could be special treatment for this case as well.

So I think that question is - do we anticipate other sources in the future. But I think that moving to `SourceFileType` can allow adding other source in the future while making the naming clearer, so we can enjoy both.